src/libponyrt/actor/actor.c

```diff
@@ -19,31 +19,148 @@
 #define PONY_SCHED_BATCH 100
 
 // Ignore padding at the end of the type.
-pony_static_assert((offsetof(pony_actor_t, gc) + sizeof(gc_t) + sizeof(size_t)) ==
+pony_static_assert((offsetof(pony_actor_t, gc) + sizeof(gc_t)) ==
    sizeof(pony_actor_pad_t), "Wrong actor pad size!");
 
 static bool actor_noblock = false;
 
-// The flags of a given actor cannot be mutated from more than one actor at
+enum
+{
+  FLAG_BLOCKED = 1 << 0,
+  FLAG_BLOCKED_SENT = 1 << 1,
+  FLAG_SYSTEM = 1 << 2,
+  FLAG_UNSCHEDULED = 1 << 3,
+  FLAG_CD_CONTACTED = 1 << 4,
+};
+
+enum
+{
+  SYNC_FLAG_PENDINGDESTROY = 1 << 0,
+  SYNC_FLAG_OVERLOADED = 1 << 1,
+  SYNC_FLAG_UNDER_PRESSURE = 1 << 2,
+  SYNC_FLAG_MUTED = 1 << 3,
+};
+
+// The sync flags of a given actor cannot be mutated from more than one actor at
 // once, so these operations need not be atomic RMW.
+static bool has_sync_flag_any(pony_actor_t* actor, uint8_t check_flags)
+{
+  uint8_t flags = atomic_load_explicit(&actor->sync_flags, memory_order_acquire);
+  return (flags & check_flags) != 0;
+}
+
+static bool has_sync_flag(pony_actor_t* actor, uint8_t flag)
+{
+  return has_sync_flag_any(actor, flag);
+}
+
+static void set_sync_flag(pony_actor_t* actor, uint8_t flag)
+{
+  uint8_t flags = atomic_load_explicit(&actor->sync_flags, memory_order_acquire);
```

Comment:
were all the atomics based on what we were doing for is_muted before?

Reply:
im looking at the atomics and they are different for sync vs what is_muted was. you appear to have weakened this for all the sync ones, I'm not sure about the flags.